gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -473,8 +473,9 @@ function roleselection.GetSelectableRolesList(maxPlys, rolesAmountList)
 	hook.Run("TTT2ModifySelectableRoles", selectableRoles)
 
 	roleselection.selectableRoles = selectableRoles
+	local tmpSelectableRoles = table.Copy(selectableRoles)
 
-	return selectableRoles
+	return tmpSelectableRoles
```

Comment:
Can be combined

Reply:
Then you've to edit the first lines of this func as well i guess

Reply:
Oh yeah, didnt see that, changed it now.